File: BinarySearch.java
//Binary Search using function
/*
Binary Search used for Searching elements in sorted arrays.
 */
import java.util.Scanner;
public class BinarySearch {
    static int Search(int[] nums, int target){
        int start = 0; //start index
        int end = nums.length-1;//ending element index
        while(start <= end){
            int mid = start + (end - start)/2;
            if(target < nums[mid])
                end = mid -1;
            else if(target > nums[mid])
                start = mid + 1;
            else
                return mid;
        }
        return -1;
    }// end of BinarySearch Function
    //main
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int target = s.nextInt();
        s.close();
        int[] nums = {-23, -21, -4, 0, 3, 43, 566, 343, 6345};
        int ans = Search(nums, target);
        System.out.println(ans);
    }
}

File: Add Code Here/cpp/binary search.cpp
#include<iostream>
using namespace std;
int binarysearch(int a[],int n,int key){
	int s=0;
	int e=n-1;
	
	while(s<=e){
		int mid=(s+e)/2;
		
		if(a[mid]==key){
			return mid;
			
		}
		else if(a[mid]>key){
			e=mid-1;
		}
		else{
			s=mid+1;
		}
		
	}
	return -1;
	
}
int main(){
	int n,key;
	cout<<"enter the no. of elements in an array:"<<endl;
	cin>>n;
	int a[1000];
	
	for(int i=0;i<n;i++){
		cin>>a[i];
	}
	cout<<"enter the value to be found"<<endl;
	cin>>key;
	cout<<binarysearch(a,n,key)<<endl;
	return 0;
	
	
}
